sympy/physics/control/lti.py

```diff
@@ -1422,3 +1424,359 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
+    >>> pprint(_, use_unicode=False)
+    [  a + s        -3     ]
+    [----------    -----   ]
+    [ 2            s + 2   ]
+    [s  + s + 1            ]
+    [                      ]
+    [    12        -a - s  ]
+    [  -----     ----------]
+    [  s + 2      2        ]
+    [            s  + s + 1]
+    [                      ]
+    [    3          -12    ]
+    [  -----       -----   ]
+    [  s + 2       s + 2   ]
+    >>> pprint(tfm_2, use_unicode=False) # State of tfm_2 is unchanged after substitution
+    [   a + s           -3       ]
+    [ ----------       -----     ]
+    [  2               s + 2     ]
+    [ s  + s + 1                 ]
+    [                            ]
+    [ 4                          ]
+    [p  - 3*p + 2      -a - s    ]
+    [------------    ----------  ]
+    [   p + s         2          ]
+    [                s  + s + 1  ]
+    [                            ]
+    [                 4          ]
+    [     3        - p  + 3*p - 2]
+    [   -----      --------------]
+    [   s + 2          p + s     ]
+
+    ``subs()`` also supports multiple substitutions.
+
+    >>> tfm_2.subs({p: 2, a: 1})  # substituting p with 2 and a with 1
+    TransferFunctionMatrix(((TransferFunction(s + 1, s**2 + s + 1, s), TransferFunction(-3, s + 2, s)), (TransferFunction(12, s + 2, s), TransferFunction(-s - 1, s**2 + s + 1, s)), (TransferFunction(3, s + 2, s), TransferFunction(-12, s + 2, s))))
+    >>> pprint(_, use_unicode=False)
+    [  s + 1        -3     ]
+    [----------    -----   ]
+    [ 2            s + 2   ]
+    [s  + s + 1            ]
+    [                      ]
+    [    12        -s - 1  ]
+    [  -----     ----------]
+    [  s + 2      2        ]
+    [            s  + s + 1]
+    [                      ]
+    [    3          -12    ]
+    [  -----       -----   ]
+    [  s + 2       s + 2   ]
+
+    See Also
+    ========
+
+    TransferFunction, Series, Parallel, Feedback
+
+    """
+    def __new__(cls, arg):
+
+        expr_mat_arg = []
+        var = arg[0][0].var
+        for row in range(len(arg)):
+            temp = []
+            for col in range(len(arg[row])):
+                if not isinstance(arg[row][col], (TransferFunction, Series, Parallel)):
+                    raise TypeError("Incompatible type found as the element of TransferFunctionMatrix")
+
+                if var != arg[row][col].var:
+                    raise ValueError("Conflicting value(s) found for `var`. All TransferFunction instances in "
+                            "TransferFunctionMatrix should use the same complex variable in Laplace domain.")
+
+                if isinstance(arg[row][col], TransferFunction):
+                    temp.append(arg[row][col].to_expr())
+                else:
+                    temp.append(arg[row][col].doit().to_expr())
```

Comment:
Why are we calling `doit()` here? In my previous PR, I defined a `doit()` method in TFM class which:
```
        Returns the resultant transfer function matrix obtained after evaluating
        the transfer functions in series or parallel configurations (if any present)
        for all entries in a transfer function matrix.
```
Now it does simplify things, but is this the right way of "doing" it?

Reply:
> Why are we calling `doit()` here? In my previous PR, I defined a `doit()` method in TFM class which:
> 
> ```
>         Returns the resultant transfer function matrix obtained after evaluating
>         the transfer functions in series or parallel configurations (if any present)
>         for all entries in a transfer function matrix.
> ```
> 
> Now it does simplify things, but is this the right way of "doing" it?

I am calling `doit` here for the formation of the `obj._expr_mat`. The actual state of the `Series` or `Parallel` element passed is not changed. 

https://github.com/sympy/sympy/blob/7a6c86302e7fc02cdcfabfee733dc17c5d34cdbc/sympy/physics/control/lti.py#L1661-L1663

Reply:
I got your point. I have removed `doit()`.